### PR TITLE
[WIP]add HEAD to distcc

### PR DESCRIPTION
**Pull request by yurikoles**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [ ] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

Please help me resolve issue that is also reported to upstream https://github.com/distcc/distcc/issues/197

Full log is [here](https://gist.github.com/c67ab0f44aa1239e3c4cda34b17dcaa9)

### CODE REVIEW

**Comment by yurikoles** [comment]
Please help me resolve issue that is also reported to upstream https://github.com/distcc/distcc/issues/197

File: Formula/distcc.rb [hunk 1]
```diff
@@ -21,12 +25,24 @@ class Distcc < Formula
 
   depends_on PythonWithoutPPCRequirement
 
+  depends_on "autoconf" if build.head?
```
Comment (MikeMcQuaid) [inline Formula/distcc.rb]: Add these all to a `head do` block.

File: Formula/distcc.rb [hunk 2]
```diff
@@ -21,12 +25,24 @@ class Distcc < Formula
...
     # Make sure python stuff is put into the Cellar.
     # --root triggers a bug and installs into HOMEBREW_PREFIX/lib/python2.7/site-packages instead of the Cellar.
-    inreplace "Makefile.in", '--root="$$DESTDIR"', ""
 
-    system "./configure", "--prefix=#{prefix}"
+    if build.head?
+      system "./autogen.sh"
+    end
+
+    inreplace "Makefile.in", '--root="$$DESTDIR"', ""
```
Comment (MikeMcQuaid) [inline Formula/distcc.rb]: Please keep this with the comment.

File: Formula/distcc.rb [hunk 3]
```diff
@@ -21,12 +25,24 @@ class Distcc < Formula
...
+    args = %W[
+      --prefix=#{prefix}
+    ]
+    args << "--without-libiberty" if build.head?
```
Comment (MikeMcQuaid) [inline Formula/distcc.rb]: Was this option removed or does it not work in HEAD?

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/distcc.rb (hunk 3 above)
Comment (yurikoles) [inline Formula/distcc.rb]: This isn't working in HEAD, this library is not shipped by us in gcc and binutils, there is a number of bugs regarding this library. [Upstream recommends](https://gcc.gnu.org/onlinedocs/libiberty/Using.html) to ship it with project source code.

**Comment by bfontaine** [comment]
@yurikoles Anything new here?

**Comment by MikeMcQuaid** [comment]
Looks like https://github.com/distcc/distcc/issues/159 will 🚢 soon with the GitHub tarball for this version.

**Comment by MikeMcQuaid** [comment]
Passing on this due to comments not being addressed. We'll review a new PR with comments addressed and CI passing. Sorry and thanks anyway!

**Pull request closed without merging**